app/src/main/java/de/tum/in/tumcampusapp/api/navigatum/domain/NavigationEntity.kt

```diff
@@ -0,0 +1,54 @@
+package de.tum.`in`.tumcampusapp.api.navigatum.domain
+
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import de.tum.`in`.tumcampusapp.component.other.general.model.Recent
+import java.io.Serializable
+
+data class NavigationEntity(
+    @SerializedName("id")
+    var id: String = "",
+    @SerializedName("type")
+    var type: String = "",
+    @SerializedName("name")
+    var name: String = "",
+    @SerializedName("subtext")
+    var subtext: String = ""
+) : Serializable {
+
+    fun getFormattedName(): String {
+        return removeHighlight(name)
+    }
```

Comment:
```suggestion
    @SerializedName("subtext")
    var subtext: String = "",
    @SerializedName("parsed_id")
    var parsedId: String? = null,
) : Serializable {

    fun getFormattedName(): String {
        return if (parsedId === null) {
            removeHighlight(name)
        } else {
            removeHighlight(parsedId!!) + " ➤ " + removeHighlight(name)
        }
    }
```

I think it would be helpful to add the `parsed_id` field as well. This way it would look like this:

<img src="https://user-images.githubusercontent.com/7429408/183260283-399bfb7e-5ada-408a-93a9-0c93392eb6fc.png" width="400" />

We could also use a different arrow symbol, e.g. `»`:

<img src="https://user-images.githubusercontent.com/7429408/183260328-e966abb5-063d-4355-93fc-190dda137f2c.png" width="400" />

If you think this makes the search results too long, I would suggest showing only the `parsed_id`, because it's only included if we consider it more helpful than the normal name.

(You can also change the sources, I've never written Kotlin so far)

Reply:
I have added ➤ option